gson/src/main/java/com/google/gson/reflect/TypeToken.java

```diff
@@ -379,27 +379,19 @@ public static <T> TypeToken<T> get(Class<T> type) {
    * <p>If {@code rawType} is a non-generic class and no type arguments are provided, this method
    * simply delegates to {@link #get(Class)} and creates a {@code TypeToken(Class)}.
    *
-   * @throws IllegalArgumentException If {@code rawType} is not of type {@code Class}, or if the
-   *     type arguments are invalid for the raw type
+   * @throws IllegalArgumentException If the type arguments are invalid for the raw type
    */
-  public static TypeToken<?> getParameterized(Type rawType, Type... typeArguments) {
+  public static TypeToken<?> getParameterized(Class<?> rawType, Type... typeArguments) {
```

Comment:
Oh, we can't change the signature of this method. It is part of the public API and it is _not_ internal. So I think you want to undo the changes in this method, and just change `rawType` to `rawClass` in the `return` statement.